src/test/java/com/squareup/kotlinpoet/FunSpecTest.kt

```diff
@@ -199,6 +216,22 @@ class FunSpecTest {
       |""".trimMargin())
   }
 
+  @Test fun functionWithReturnKDoc() {
+    val unitType = UNIT
+    val funSpec = FunSpec.builder("foo")
+        .addParameter(ParameterSpec.builder("f", LambdaTypeName.get(returnType = unitType)).build())
+        .returns(String::class, CodeBlock.builder().add("@return the foo.\n").build())
```

Comment:
Also, can we handle the `\n`? Ideally `"the foo"` should be a valid KDoc string, which we'll print out as `@return a foo\n`

Reply:
Sure I don't mind doing that. However, when we print out `@param` comments, we don't add the new line by default and that's why I didn't add the line either. I like the idea of not having to manually append the `\n`, but for the sake of consistency we'd want to add it to the `@param` comments also, which might be a breaking change. What do you think?

Reply:
Sorry, missed this comment before my recent review... I'm not sure what would the result be if `\n` is not passed in as a part of the string, but I guess it's not something somebody would ever want, hence library users will pass in a string without the new line, notice that it looks ugly and will hack the library by appending the `\n` - this sucks. Let's add `\n` ourselves, and I'll open an issue to fix it for params.